x/foundation/keeper/internal/genesis.go

```diff
@@ -48,6 +48,9 @@ func (k Keeper) InitGenesis(ctx sdk.Context, data *foundation.GenesisState) erro
 
 	k.SetPool(ctx, data.Pool)
 
+	// init module accounts just in case
+	_ = k.authKeeper.GetModuleAccount(ctx, foundation.ModuleName)
```

Comment:
In case of absence of the permission, it will return nil, hence it should check the result, and if necessary, panic.